db/virtual_table.hh

```diff
@@ -55,4 +56,19 @@ public:
     void set_database(database& db) { _db = &db; }
 };
 
+// Produces results by filling a memtable on each read.
+// Use when the amount of data is not significant relative to shard's memory size.
+class memtable_filling_virtual_table : public virtual_table {
+public:
+    using virtual_table::virtual_table;
+
+    // Override one of these execute() overloads.
+    // The handler is always allowed to produce more data than implied by the query_restrictions.
+    virtual future<> execute(memtable&) { return make_ready_future<>(); }
```

Comment:
Why is there a version without the permit update? All readers should update the permit.

Reply:
True, we don't that anymore

Reply:
got rid of it, done